cmd/reindex/main.go

```diff
@@ -0,0 +1,314 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ONSdigital/dp-api-clients-go/v2/zebedee"
+	dpEs "github.com/ONSdigital/dp-elasticsearch/v3"
+	dpEsClient "github.com/ONSdigital/dp-elasticsearch/v3/client"
+	v710 "github.com/ONSdigital/dp-elasticsearch/v3/client/elasticsearch/v710"
+	"github.com/ONSdigital/dp-net/v2/awsauth"
+	dphttp2 "github.com/ONSdigital/dp-net/v2/http"
+	"github.com/ONSdigital/dp-search-api/elasticsearch"
+	extractorModels "github.com/ONSdigital/dp-search-data-extractor/models"
+	importerModels "github.com/ONSdigital/dp-search-data-importer/models"
+	"github.com/ONSdigital/dp-search-data-importer/transform"
+)
+
+var (
+	maxConcurrentExtractions = 20
+	maxConcurrentIndexings   = 30
+)
+
+type cliConfig struct {
+	aws          AWSConfig
+	zebedeeURL   string
+	esURL        string
+	signRequests bool
+}
+
+type AWSConfig struct {
+	filename              string
+	profile               string
+	region                string
+	service               string
+	tlsInsecureSkipVerify bool
+}
+
+type zebedeeClient interface {
+	GetPublishedIndex(ctx context.Context, piRequest *zebedee.PublishedIndexRequestParams) (zebedee.PublishedIndex, error)
+	GetPublishedData(ctx context.Context, uriString string) ([]byte, error)
+}
+
+type Document struct {
+	URI  string
+	Body []byte
+}
+
+func main() {
+	fmt.Printf("Hola %s!\n", Name)
+
+	ctx := context.Background()
+	cfg := getConfig(ctx)
+
+	hcClienter := dphttp2.NewClient()
+	hcClienter.SetMaxRetries(2)
+	hcClienter.SetTimeout(30 * time.Second) // Published Index takes about 10s to return so add a bit more
+	zebClient := zebedee.NewClientWithClienter(cfg.zebedeeURL, hcClienter)
+
+	esHTTPClient := hcClienter
+	if cfg.signRequests {
+		fmt.Println("Use a signing roundtripper client")
+		awsSignerRT, err := awsauth.NewAWSSignerRoundTripper(cfg.aws.filename, cfg.aws.filename, "eu-west-1", "es",
+			awsauth.Options{TlsInsecureSkipVerify: true})
```

Comment:
I dont think the awsauth.Options field `TlsInsecureSkipVerify` should be set; I think this should be false; or better yet configurable

Reply:
So yes it is set in local.go; but should also be set in aws.go file (hardcoded to false) and change here to use cfg.AWS.TLSInsecureSkipVerify

Reply:
Done in update 1.